client/ui/components/AddonGroupForm.jsx

```diff
@@ -0,0 +1,246 @@
+import React, { Component, PropTypes as pt } from 'react';
+import { browserHistory } from 'react-router';
+import { Field, reduxForm } from 'redux-form';
+import { TextField } from 'redux-form-material-ui';
+
+import Chip from 'material-ui/Chip';
+import FlatButton from 'material-ui/FlatButton';
+import LinearProgress from 'material-ui/LinearProgress';
+import MenuItem from 'material-ui/MenuItem';
+import RaisedButton from 'material-ui/RaisedButton';
+import SelectField from 'material-ui/SelectField';
+import NavigationChevronLeft from 'material-ui/svg-icons/navigation/chevron-left';
+import { Toolbar, ToolbarGroup, ToolbarSeparator } from 'material-ui/Toolbar';
+
+import ErrorSnackbar from './stateless/ErrorSnackbar.jsx';
+import FetchErrorList from './stateless/FetchErrorList.jsx';
+import LoadingIndicator from './stateless/LoadingIndicator.jsx';
+
+
+const style = {
+  addonChip: {
+    marginRight: '5px',
+  },
+  addonChips: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  toolbar: {
+    justifyContent: 'flex-end',
+  },
+  refresh: {
+    display: 'inline-block',
+    position: 'relative',
+  },
+};
+
+class AddonSelectField extends Component {
+  static propTypes = {
+    addons: pt.array.isRequired,
+    floatingLabelText: pt.string.isRequired,
+    input: pt.object.isRequired,
+    name: pt.string.isRequired,
+  };
+
+  constructor() {
+    super();
+
+    // Bind `this` to event handlers and render functions
+    this.handleChange = this.handleChange.bind(this);
+    this.handleRequestDelete = this.handleRequestDelete.bind(this);
+    this.renderAddonChips = this.renderAddonChips.bind(this);
+    this.renderAddonItems = this.renderAddonItems.bind(this);
+  }
+
+  handleChange(event, index, addonId) {
+    const { onChange, value } = this.props.input;
+    onChange([...value, addonId]);
+  }
+
+  handleRequestDelete(addonId) {
+    const { onChange, value } = this.props.input;
+    onChange(value.filter(v => v !== addonId));
+  }
+
+  renderAddonItems() {
+    const { addons, input } = this.props;
+    const items = [];
+
+    addons.forEach(addon => {
+      if (input.value.indexOf(addon.id) === -1) {
+        items.push(<MenuItem key={addon.id} value={addon.id} primaryText={addon.name} />);
+      }
+    });
+
+    return items;
+  }
+
+  renderAddonChips() {
+    const { addons, input } = this.props;
+    const items = [];
+
+    input.value.forEach(addonId => {
+      const addon = addons.find(a => a.id === addonId);
+
+      items.push(
+        <Chip
+          key={addonId}
+          style={style.addonChip}
+          onRequestDelete={() => this.handleRequestDelete(addonId)}
+        >
+          {addon.name}
```

Comment:
I think you should include the version number here too (and in the `MenuItem`s).